streams/src/main/java/org/apache/kafka/streams/kstream/internals/GroupedStreamAggregateBuilder.java

```diff
@@ -84,7 +84,7 @@
         if (repartitionRequired) {
             final OptimizableRepartitionNodeBuilder<K, V> repartitionNodeBuilder = optimizableRepartitionNodeBuilder();
             final String repartitionTopicPrefix = userProvidedRepartitionTopicName != null ? userProvidedRepartitionTopicName : storeBuilder.name();
-            sourceName = createRepartitionSource(repartitionTopicPrefix, repartitionNodeBuilder);
+            sourceName = createRepartitionSource(repartitionTopicPrefix, repartitionNodeBuilder, !repartitionTopicPrefix.contains("KSTREAM-"));
```

Comment:
We need to have this check here as if the user doesn't provide a name via `Grouped`, we'll use the name of the store which the user could provide or the store name can be generated as well.